### ISSUE TITLE
# tflint-ignore: terraform_unused_declarations no longer masks a finding

### ISSUE THREAD

**Issue by richardj-bsquare** [terraform-linters/tflint#1873]
### Summary

I'm having problems ignoring unused declarations. I've got these kinds of statements scattered through various input variables. And at some point, they were working, but now they *always* report failures. 

Regression testing 0.46.0 and 0.47.0 shows these statements working just fine, so the problem appeared for me in 0.48.0

Perhaps related to: https://github.com/terraform-linters/tflint/pull/1834 ?

### Command

tflint

### Terraform Configuration

```terraform
variable "ecr_registry_id" {
  # tflint-ignore: terraform_unused_declarations
  description = "AWS ECR registry ID from which to fetch ECR container images (deprecated)"
  type        = string
  default     = "deprecated_ecr_registry_id"
}
```
```


### TFLint Configuration

```HCL
plugin "terraform" {
  enabled = true
  preset  = "all"
}

rule "terraform_naming_convention" {
  enabled = false
}
```

```
$ tflint --version
TFLint version 0.48.0
+ ruleset.terraform (0.4.0-bundled)
```
```


### Output

```Shell
Warning: [Fixable] variable "ecr_registry_id" is declared but not used (terraform_unused_declarations)

  on variables.tf line 65:
  65: variable "ecr_registry_id" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.4.0/docs/rules/terraform_unused_declarations.md
```
```


### TFLint Version

0.48.0

### Terraform Version

Terraform v1.5.7 on darwin_arm64

### Operating System

- [ ] Linux
- [X] macOS
- [ ] Windows

**Comment by bendrucker** [comment]
The annotation is misplaced, this should never have worked. Annotations should appear on the line before the relevant range. As in above the variable. 

If that's not it please fix the formatting of the issue and reopen.

**Comment by richardj-bsquare** [comment]
I've tried the annotation absolutely everywhere before the line and even multiple times, nothing seems to work now, and was working before.

Tested this:
```
# tflint-ignore: terraform_unused_declarations
variable "ecr_registry_id" {
  description = "AWS ECR registry ID from which to fetch ECR container images (deprecated)"
  type        = string
  default     = "deprecated_ecr_registry_id"
}
```
Doesn't work on 0.48.0, works on previous versions...
